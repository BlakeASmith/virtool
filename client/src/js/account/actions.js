/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */
import {
    WS_UPDATE_ACCOUNT,
    GET_ACCOUNT,
    UPDATE_ACCOUNT_SETTINGS,
    CREATE_API_KEY,
    UPDATE_API_KEY,
    REMOVE_API_KEY,
    LOGOUT
} from "../actionTypes";


export function wsUpdateAccount (data) {
    return {
        type: WS_UPDATE_ACCOUNT,
        data
    };
}

export function getAccount () {
    return {
        type: GET_ACCOUNT.REQUESTED
    }
}

export function updateAccountSettings (update) {
    return {
        type: UPDATE_ACCOUNT_SETTINGS.REQUESTED,
        update
    };
}

export function createAPIKey (name, permissions, callback) {
    return {
        type: CREATE_API_KEY.REQUESTED,
        name,
        permissions,
        callback
    };
}

export function updateAPIKey (keyId, permissions) {
    return {
        type: UPDATE_API_KEY.REQUESTED,
        keyId,
        permissions
    };
}

export function removeAPIKey (keyId) {
    return {
        type: REMOVE_API_KEY.REQUESTED,
        keyId
    };
}

export function logout () {
    return {
        type: LOGOUT.REQUESTED
    }
}
