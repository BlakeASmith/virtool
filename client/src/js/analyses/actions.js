import {
    WS_INSERT_ANALYSIS,
    WS_UPDATE_ANALYSIS,
    WS_REMOVE_ANALYSIS,
    FIND_ANALYSES,
    GET_ANALYSIS,
    ANALYZE,
    BLAST_NUVS,
    REMOVE_ANALYSIS,
    CLEAR_ANALYSIS,
    COLLAPSE_ANALYSIS,
    SET_PATHOSCOPE_FILTER,
    TOGGLE_SORT_PATHOSCOPE_DESCENDING,
    TOGGLE_SHOW_PATHOSCOPE_READS,
    SET_ACTIVE_HIT_ID,
    TOGGLE_FILTER_ORFS,
    TOGGLE_FILTER_SEQUENCES,
    SET_SEARCH_IDS,
    SET_ANALYSIS_SORT_KEY,
    TOGGLE_RESULT_EXPANDED,
    CLEAR_ANALYSES
} from "../app/actionTypes";
import { simpleActionCreator } from "../utils/utils";

export const wsInsertAnalysis = data => ({
    type: WS_INSERT_ANALYSIS,
    data
});

/**
 * Returns an action that should be dispatched when a analysis document is updated via websocket.
 *
 * @func
 * @param data {object} update data passed in the websocket message
 * @returns {object}
 */
export const wsUpdateAnalysis = data => ({
    type: WS_UPDATE_ANALYSIS,
    data
});

/**
 * Returns an action that should be dispatched when a analysis document is removed via websocket.
 *
 * @func
 * @param data {string} the id for the specific analysis
 * @returns {object}
 */
export const wsRemoveAnalysis = data => ({
    type: WS_REMOVE_ANALYSIS,
    data
});

export const collapseAnalysis = simpleActionCreator(COLLAPSE_ANALYSIS);

export const setActiveHitId = id => ({
    type: SET_ACTIVE_HIT_ID,
    id
});

export const setSearchIds = ids => ({
    type: SET_SEARCH_IDS,
    ids
});

export const setAnalysisSortKey = sortKey => ({
    type: SET_ANALYSIS_SORT_KEY,
    sortKey
});

export const setPathoscopeFilter = key => ({
    type: SET_PATHOSCOPE_FILTER,
    key
});

export const toggleFilterORFs = simpleActionCreator(TOGGLE_FILTER_ORFS);

export const toggleFilterSequences = simpleActionCreator(TOGGLE_FILTER_SEQUENCES);

export const togglePathoscopeSortDescending = simpleActionCreator(TOGGLE_SORT_PATHOSCOPE_DESCENDING);

export const toggleResultExpanded = id => ({
    type: TOGGLE_RESULT_EXPANDED,
    id
});

export const toggleShowPathoscopeReads = simpleActionCreator(TOGGLE_SHOW_PATHOSCOPE_READS);

export const findAnalyses = (sampleId, term, page) => ({
    type: FIND_ANALYSES.REQUESTED,
    sampleId,
    term,
    page
});

/**
 * Returns action that can trigger an API call for retrieving a specific analysis.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @returns {object}
 */
export const getAnalysis = analysisId => ({
    type: GET_ANALYSIS.REQUESTED,
    analysisId
});

export const clearAnalyses = simpleActionCreator(CLEAR_ANALYSES);
export const clearAnalysis = simpleActionCreator(CLEAR_ANALYSIS);

/**
 * Returns action that can trigger an API call for sample analysis.
 *
 * @func
 * @param sampleId {string} unique sample id
 * @param refId {string} unique id for a reference
 * @param algorithm {string} algorithm type
 * @param userId {string} the id of the requesting user
 * @returns {object}
 */
export const analyze = (sampleId, refId, algorithm, subtractionId, userId) => ({
    type: ANALYZE.REQUESTED,
    algorithm,
    refId,
    sampleId,
    subtractionId,
    userId
});

/**
 * Returns action that can trigger an API call for BLASTing NuV analysis contigs.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @param sequenceIndex {number} index of the sequence
 * @returns {object}
 */
export const blastNuvs = (analysisId, sequenceIndex) => ({
    type: BLAST_NUVS.REQUESTED,
    analysisId,
    sequenceIndex
});

/**
 * Returns action that can trigger an API call for removing an analysis.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @returns {object}
 */
export const removeAnalysis = analysisId => ({
    type: REMOVE_ANALYSIS.REQUESTED,
    analysisId
});
