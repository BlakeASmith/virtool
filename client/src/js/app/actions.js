import { PUSH_STATE } from "./actionTypes";

export const pushState = state => ({
    type: PUSH_STATE,
    state
});
