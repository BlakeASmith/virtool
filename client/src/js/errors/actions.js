/**
 * Redux action creator for clearing errors.
 *
 * @module error/actions
 */
import { CLEAR_ERROR } from "../actionTypes";

/**
 * Clears specific temprorarily stored error.
 *
 * @func
 * @returns {object}
 */
export const clearError = (error) => ({
    type: CLEAR_ERROR,
    error
});
