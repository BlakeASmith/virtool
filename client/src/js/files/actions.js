/**
 * Redux action creators for use with file management.
 *
 * @module files/actions
 */

import { simpleActionCreator } from "../utils";
import {
    WS_UPDATE_FILE,
    WS_REMOVE_FILE,
    FIND_FILES,
    REMOVE_FILE,
    UPLOAD,
    UPLOAD_PROGRESS,
    HIDE_UPLOAD_OVERLAY
} from "../actionTypes";

/**
 * Returns an action that should be dispatched when a file document is updated via websocket.
 *
 * @func
 * @param data {object} the data passed in the websocket message
 * @returns {object}
 */
export const wsUpdateFile = (data) => ({
    type: WS_UPDATE_FILE,
    data
});

/**
 * Returns an action that should be dispatched when a file document is removed via websocket.
 *
 * @func
 * @param data {object} the data passed in the websocket message
 * @returns {object}
 */
export const wsRemoveFile = (data) => ({
    type: WS_REMOVE_FILE,
    data
});

/**
 * Returns an action that can trigger an API call that finds file documents.
 *
 * @func
 * @param fileType {string} a file type to filter the returned documents by
 * @param page {number} which page of results to return
 * @returns {object}
 */
export const findFiles = (fileType, page) => ({
    type: FIND_FILES.REQUESTED,
    fileType,
    page
});

/**
 * Returns an action that can trigger the upload of a file to the server.
 *
 * @param localId
 * @param file
 * @param fileType
 * @param onProgress
 * @returns {{type: string, localId: *, file: *, fileType: *, onProgress: *}}
 */
export const upload = (localId, file, fileType, onProgress) => ({
    type: UPLOAD.REQUESTED,
    localId,
    file,
    fileType,
    onProgress
});

export const removeFile = (fileId) => ({
    type: REMOVE_FILE.REQUESTED,
    fileId
});

export const uploadProgress = (localId, progress) => ({
    type: UPLOAD_PROGRESS,
    localId,
    progress
});

export const hideUploadOverlay = simpleActionCreator(HIDE_UPLOAD_OVERLAY);
