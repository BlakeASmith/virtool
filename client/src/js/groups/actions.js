import { simpleActionCreator } from "../utils";
import { LIST_GROUPS, CREATE_GROUP, SET_GROUP_PERMISSION, REMOVE_GROUP } from "../actionTypes";

export const listGroups = simpleActionCreator(LIST_GROUPS.REQUESTED);

export const createGroup = (groupId) => ({
    type: CREATE_GROUP.REQUESTED,
    groupId
});

export const setGroupPermission = (groupId, permission, value) => ({
    type: SET_GROUP_PERMISSION.REQUESTED,
    groupId,
    permission,
    value
});

export const removeGroup = (groupId) => ({
    type: REMOVE_GROUP.REQUESTED,
    groupId
});
