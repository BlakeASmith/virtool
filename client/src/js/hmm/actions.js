/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */
import { FIND_HMMS, GET_HMM, INSTALL_HMMS } from "../actionTypes";

export const findHMMs = (term, page) => {
    return {
        type: FIND_HMMS.REQUESTED,
        term,
        page
    };
};

export const getHmm = (hmmId) => {
    return {
        type: GET_HMM.REQUESTED,
        hmmId: hmmId
    };
};

export const installHMMs = () => {
    return {
        type: INSTALL_HMMS.REQUESTED
    }
};
