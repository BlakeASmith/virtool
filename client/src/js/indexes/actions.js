import { simpleActionCreator } from "../utils";
import {
    FIND_INDEXES,
    GET_INDEX,
    GET_UNBUILT,
    CREATE_INDEX,
    GET_INDEX_HISTORY,
    SHOW_REBUILD,
    HIDE_REBUILD
} from "../actionTypes";

export const findIndexes = simpleActionCreator(FIND_INDEXES.REQUESTED);

export const getIndex = (indexVersion) => ({
    type: GET_INDEX.REQUESTED,
    indexVersion
});

export const getUnbuilt = simpleActionCreator(GET_UNBUILT.REQUESTED);

export const createIndex = simpleActionCreator(CREATE_INDEX.REQUESTED);

export const getIndexHistory = (indexVersion) => ({
    type: GET_INDEX_HISTORY.REQUESTED,
    indexVersion
});

export const showRebuild = () => simpleActionCreator(SHOW_REBUILD);

export const hideRebuild = simpleActionCreator(HIDE_REBUILD);
