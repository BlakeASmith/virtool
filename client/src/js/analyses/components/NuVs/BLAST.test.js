import * as actions from "../../actions";
import NuVsBLASTContainer, { BLASTInProgress, BLASTResults, NuVsBLAST } from "./BLAST";

describe("<NuVsBLAST />", () => {
    let props;
    let wrapper;

    it("renders correctly", () => {
        props = {
            blast: {
                ready: true,
                result: {
                    hits: [
                        {
                            name: "test",
                            accession: "NC123",
                            evalue: 3,
                            score: 1,
                            identity: 2,
                            align_len: 4
                        }
                    ]
                }
            },
            analysisId: "123abc"
        };
        wrapper = shallow(<NuVsBLAST {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders correctly with no BLAST hits", () => {
        props = {
            analysisId: "123abc",
            blast: { ready: true, result: { hits: [] } }
        };
        wrapper = shallow(<NuVsBLAST {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders correctly with in progress BLAST", () => {
        props = {
            analysisId: "123abc",
            blast: {
                ready: false,
                interval: 5,
                last_checked_at: "2018-02-14T17:12:00.000000Z",
                rid: "ABC123"
            }
        };
        wrapper = shallow(<NuVsBLAST {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders alert when no blast data available", () => {
        props = { analysisId: "123abc", blast: null };
        wrapper = shallow(<NuVsBLAST {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("renders <BLASTInProgress /> subcomponent", () => {
        props = {
            interval: 5,
            lastCheckedAt: "2018-02-14T17:12:00.000000Z",
            rid: "ABC123"
        };
        wrapper = shallow(<BLASTInProgress {...props} />);
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({ rid: null });
        expect(wrapper).toMatchSnapshot();
    });

    it("renders <BLASTResults /> component", () => {
        props = {
            hits: [
                {
                    name: "test",
                    accession: "NC123",
                    evalue: 3,
                    score: 1,
                    identity: 2,
                    align_len: 4
                }
            ]
        };
        wrapper = shallow(<BLASTResults {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("Clicking BLAST button dispatches blastNuvs() action", () => {
        const spy = sinon.spy(actions, "blastNuvs");
        expect(spy.called).toBe(false);

        const initialState = { analyses: { detail: { id: "123abc" } } };
        const store = mockStore(initialState);
        props = { blast: null, sequenceIndex: 0 };

        wrapper = shallow(<NuVsBLASTContainer store={store} {...props} />).dive();
        wrapper.find({ icon: "cloud" }).prop("onClick")();
        expect(spy.calledWith("123abc", 0)).toBe(true);

        spy.restore();
    });
});
