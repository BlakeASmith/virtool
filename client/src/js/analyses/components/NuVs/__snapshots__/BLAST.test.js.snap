// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NuVsBLAST /> renders <BLASTInProgress /> subcomponent 1`] = `
<Uncontrolled(Panel)>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    <Flex
      alignContent="stretch"
      alignItems="center"
      direction="row"
      justifyContent="flex-start"
      wrap="nowrap"
    >
      <FlexItem
        alignSelf={null}
        basis="auto"
        grow={0}
        shrink={1}
      >
        <ClipLoader
          color="#000"
          loading={true}
          size={16}
        />
      </FlexItem>
      <FlexItem
        alignSelf={null}
        basis="auto"
        grow={0}
        pad={5}
        shrink={1}
      >
        <span>
          BLAST in progress 
           with RID 
        </span>
        <a
          href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?    CMD=Web&PAGE_TYPE=BlastFormatting&OLD_BLAST=false&GET_RID_INFO=on&RID=ABC123"
          rel="noopener noreferrer"
          target="_blank"
        >
          ABC123
           
          <sup>
            <Icon
              faStyle="fas"
              fixedWidth={false}
              name="new-tab"
              pullRight={false}
            />
          </sup>
        </a>
      </FlexItem>
      <FlexItem
        alignSelf={null}
        basis="auto"
        grow={1}
        shrink={1}
      >
        <small
          className="pull-right"
        >
          Last checked 
          <RelativeTime
            time="2018-02-14T17:12:00.000000Z"
          />
          . Checking again in 
          8 months ago
          .
        </small>
      </FlexItem>
    </Flex>
  </PanelBody>
</Uncontrolled(Panel)>
`;

exports[`<NuVsBLAST /> renders <BLASTInProgress /> subcomponent 2`] = `
<Uncontrolled(Panel)>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    <Flex
      alignContent="stretch"
      alignItems="center"
      direction="row"
      justifyContent="flex-start"
      wrap="nowrap"
    >
      <FlexItem
        alignSelf={null}
        basis="auto"
        grow={0}
        shrink={1}
      >
        <ClipLoader
          color="#000"
          loading={true}
          size={16}
        />
      </FlexItem>
      <FlexItem
        alignSelf={null}
        basis="auto"
        grow={0}
        pad={5}
        shrink={1}
      >
        <span>
          BLAST in progress 
        </span>
      </FlexItem>
    </Flex>
  </PanelBody>
</Uncontrolled(Panel)>
`;

exports[`<NuVsBLAST /> renders <BLASTResults /> component 1`] = `
<Uncontrolled(Panel)>
  <PanelHeading
    bsClass="panel"
    componentClass="div"
  >
    NCBI BLAST
  </PanelHeading>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    <Table
      bordered={false}
      bsClass="table"
      condensed={true}
      hover={false}
      responsive={false}
      striped={false}
    >
      <thead>
        <tr>
          <th>
            Accession
          </th>
          <th>
            Name
          </th>
          <th>
            E-value
          </th>
          <th>
            Score
          </th>
          <th>
            Identity
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="0"
        >
          <td>
            <a
              href="https://www.ncbi.nlm.nih.gov/nuccore/NC123"
              rel="noopener noreferrer"
              target="_blank"
            >
              NC123
            </a>
          </td>
          <td>
            test
          </td>
          <td>
            3
          </td>
          <td>
            1
          </td>
          <td>
            0.50
          </td>
        </tr>
      </tbody>
    </Table>
  </PanelBody>
</Uncontrolled(Panel)>
`;

exports[`<NuVsBLAST /> renders alert when no blast data available 1`] = `
<Alert
  bsStyle="warning"
>
  <Flex
    alignContent="stretch"
    alignItems="center"
    direction="row"
    justifyContent="flex-start"
    wrap="nowrap"
  >
    <FlexItem
      alignSelf={null}
      basis="auto"
      grow={0}
      shrink={1}
    >
      <Icon
        faStyle="fas"
        fixedWidth={false}
        name="info-circle"
        pullRight={false}
      />
    </FlexItem>
    <FlexItem
      alignSelf={null}
      basis="auto"
      grow={1}
      pad={5}
      shrink={1}
    >
      This sequence has no BLAST information attached to it.
    </FlexItem>
    <FlexItem
      alignSelf={null}
      basis="auto"
      grow={0}
      pad={10}
      shrink={1}
    >
      <Button
        bsSize="small"
        bsStyle="default"
        icon="cloud"
        onClick={[Function]}
        pullRight={false}
        tipPlacement="top"
      >
        BLAST at NCBI
      </Button>
    </FlexItem>
  </Flex>
</Alert>
`;

exports[`<NuVsBLAST /> renders correctly 1`] = `
<BLASTResults
  hits={
    Array [
      Object {
        "accession": "NC123",
        "align_len": 4,
        "evalue": 3,
        "identity": 2,
        "name": "test",
        "score": 1,
      },
    ]
  }
/>
`;

exports[`<NuVsBLAST /> renders correctly with in progress BLAST 1`] = `
<BLASTInProgress
  interval={5}
  lastCheckedAt="2018-02-14T17:12:00.000000Z"
  rid="ABC123"
/>
`;

exports[`<NuVsBLAST /> renders correctly with no BLAST hits 1`] = `
<Uncontrolled(Panel)>
  <PanelHeading
    bsClass="panel"
    componentClass="div"
  >
    NCBI BLAST
  </PanelHeading>
  <PanelBody
    bsClass="panel"
    collapsible={false}
  >
    No BLAST hits found.
  </PanelBody>
</Uncontrolled(Panel)>
`;
